src/dotnet/APIView/APIViewWeb/Managers/ReviewManager.cs

```diff
@@ -496,51 +801,316 @@ private async Task<ReviewListItemModel> ToggleReviewApproval(ClaimsPrincipal use
+                        {
+                            // Use the same pipeline generation approach as API revisions
+                            await _apiRevisionsManager.RunAPIRevisionGenerationPipeline(batch, language);
+                        }
+                        catch (Exception ex)
                         {
-                            FileID = file.FileId,
-                            ReviewID = review.Id,
-                            RevisionID = revision.Id,
-                            FileName = Path.GetFileName(file.FileName)
-                        });
+                            _telemetryClient.TrackException(ex);
+                            // Continue with other batches
+                        }
                     }
                 }
+            }
+            catch (Exception ex)
+            {
+                _telemetryClient.TrackException(ex);
+                // Don't rethrow - this shouldn't break the background update process
+            }
+        }
+
+        /// <summary>
+        /// Get all pending namespace approval requests in a single optimized database query
+        /// This queries review records directly with database-level filtering for better performance
+        /// </summary>
+        /// <param name="limit">Maximum number of results to return (default 100)</param>
+        /// <returns>List of reviews that have pending namespace approval requests</returns>
+        public async Task<List<ReviewListItemModel>> GetPendingNamespaceApprovalsBatchAsync(int limit = 100)
+        {
+            var pendingApprovals = new List<ReviewListItemModel>();
 
-                // This should be changed to configurable batch count
-                if (paramList.Count >= backgroundBatchProcessCount)
+            try
+            {
+                using (var cts = new CancellationTokenSource(TimeSpan.FromSeconds(30)))
                 {
-                    _telemetryClient.TrackTrace($"Running pipeline to update reviews for {language} with batch size {paramList.Count}");
-                    await _apiRevisionsManager.RunAPIRevisionGenerationPipeline(paramList, languageService.Name);
-                    // Delay of 10 minute before starting next batch
-                    // We should try to increase the number of revisions in the batch than number of runs.
-                    await Task.Delay(600000);
-                    paramList.Clear();
+                    // Get all reviews that have namespace approval requests with single optimized query
+                    var sdkLanguages = ApiViewConstants.SdkLanguages;
+                    var allNamespaceReviews = await _reviewsRepository.GetPendingNamespaceApprovalReviewsAsync(sdkLanguages);
+
+                    // Add all reviews up to the limit
+                    pendingApprovals.AddRange(allNamespaceReviews.Take(limit));
+
+                    // Sort to show all reviews by request time
+                    var sortedApprovals = pendingApprovals
```

Comment:
Shouldn't we sort first and then take top 100 if your plan is to show most recent 100?